### DISCUSSION TITLE
AOT for Uno Hello World app

### DISCUSSION THREAD

**Discussion by andrey-shavelev** [unoplatform/uno#9457] [Q&A] [answered]
Hello Everyone,

I have problems enabling AOT for the Uno application and will apricate any advice. 

1. I created a 'Hello World' Uno app from the template using `dotnet new` (I tried both net5 and net6 templates). 
2. I enabled AOT in the project file, using these instructions: https://platform.uno/blog/build-net-aot-for-webassembly-in-visual-studio-with-uno-platform/3. 

The problem is that it does not build. The log contains errors like this:
```
  [00:00:18.0113282] [317/846] [AOT] ./linker-out/System.Collections.Concurrent.dll -> ./System.Collections.Concurrent.dll.bc.tmp 
  [00:00:18.0113853] FAILED: System.Collections.Concurrent.dll.bc.tmp 
```
I'm using macOS if it matters.

**Comment by jeromelaban** [comment, marked as answer]
Wasm AOT on macOS for .NET 6 is not yet supported unfortunately, and likely still won't be as of .NET 7. It's [not clear in the docs](https://platform.uno/docs/articles/external/uno.wasm.bootstrap/doc/runtime-execution-modes.html), though, we'll update them.

You can use a [Linux container](https://platform.uno/docs/articles/external/uno.wasm.bootstrap/doc/runtime-execution-modes.html#required-configuration-for-aot-mixed-mode-or-static-linking-on-linux) to build for AOT on macOS.

**Reply by andrey-shavelev** [reply]
Thanks for the reply, @jeromelaban. Is Ubuntu 22.04 supported?

**Reply by jeromelaban** [reply]
Yes it is.
